api/v1beta1/infraenv_types.go

```diff
@@ -98,6 +98,24 @@ type InfraEnvSpec struct {
 	// +kubebuilder:default=DiscoveryImageAlways
 	// +optional
 	IPXEScriptType IPXEScriptType `json:"ipxeScriptType"`
+
+	// The discovery kernel arguments to be passed during boot time.
+	// Applicable for both iPXE, and ISO streaming from Image Service.
+	// +optional
+	DiscoveryKernelArguments []KernelArgument `json:"discoveryKernelArguments,omitempty"`
```

Comment:
I don't think we need `Discovery` here as everything in the infra env is related to discovery.
For example, we not writing `DiscoveryIgnitionConfigOverride` here.

Reply:
Couldn't someone think that these arguments will be used after reboot?

Reply:
No? We already have an existing mechanism for that.
And if they do then they should try to understand how this system works a bit more before using it :sweat_smile:

Reply:
> No? We already have an existing mechanism for that.

Regardless, name should provide best hint to the usage.

> And if they do then they should try to understand how this system works a bit more before using it

But when someone tries to understand the code before using it, clearer names will help understanding. 

I think we should try to keep the same names between db and kube entities.  It helps during searches and it provides a hint that they represent the same thing.  I think it is clearer.

Reply:
I agree with both of you :)
Let's go with KernelArguments, and change the swagger to match (it's also in the context of infra-env there).

Reply:
@avishayt I am not sure we want to do that.
discovery_kernel_arguments is also the name of the column in the database.  Do we want to change ?  It will require migration.

Reply:
Should be straightforward, no?

Reply:
It means we keep the migration forever.
BTW, I think the name DiscoveryKernelArguments is better. 
Do you think the change is essential ?

Reply:
Change